pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -450,11 +458,22 @@ public TimeBoundaryInfo getTimeBoundaryInfo(String offlineTableName) {
     return timeBoundaryManager != null ? timeBoundaryManager.getTimeBoundaryInfo() : null;
   }
 
+  /**
+   * Returns the timeout in milliseconds for the given table, or {@code null} if the timeout is not configured in the
+   * table config.
+   */
+  @Nullable
+  public Long getTimeoutMs(String tableNameWithType) {
```

Comment:
Can we specify the timeout here? Is it the max overall timeout for each query?

Reply:
I don't quite understand this comment. This is the timeout for the table specified in the table config